test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/TaskListControllerSpec.scala

```diff
@@ -82,15 +89,120 @@ class TaskListControllerSpec
         }
 
+        "the session data indicates that they are filling in a return and the triage section is incomplete" in {
+          testTemplate(
+            sample[FillingOutReturn].copy(draftReturn = sample[DraftReturn].copy(triageAnswers =
+              sample[IncompleteIndividualTriageAnswers])),
+            TaskListStatus.InProgress
+          )
+        }
+
+        "the session data indicates that they are filling in a return and the triage section is complete" in {
+          testTemplate(
+            sample[FillingOutReturn].copy(draftReturn = sample[DraftReturn].copy(triageAnswers =
+              sample[CompleteIndividualTriageAnswers])),
+            TaskListStatus.Complete
+          )
+        }
       }
 
-    }
+      "display the page with the proper Enter property address section status" when {
+        "the session data indicates that they are filling in a return and enter property address is todo" in {
+          val fillingOutReturn = sample[FillingOutReturn].copy(draftReturn = sample[DraftReturn]
+              .copy(propertyAddress = None))
 
-  }
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              Future.successful(
+                Right(
+                  Some(
+                    SessionData.empty.copy(
+                      journeyStatus = Some(fillingOutReturn)
+                    )
+                  )
+                )
+              )
+            )
+          }
+
+          val result = performAction()
+          status(result) shouldBe OK
+
+          val doc: Document = parse(contentAsString(result))
+          doc.title shouldBe messagefromMessageKey("service.title") + " - GOV.UK"
+
+          doc.select("li#propertyAddress > a").text shouldBe messagefromMessageKey("task-list.enter-property-address.link")
+          doc.select("li#propertyAddress > a").attr("href") shouldBe address.routes.PropertyAddressController.enterPostcode().url
+          doc.select("li#propertyAddress > strong").text shouldBe TaskListStatus.ToDo.toString
+        }
+
+        "the session data indicates that they are filling in a return and enter property address is complete" in {
+          val fillingOutReturn = sample[FillingOutReturn].copy(draftReturn = sample[DraftReturn]
+            .copy(propertyAddress = Some(sample[UkAddress])))
+
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              Future.successful(
+                Right(
+                  Some(
+                    SessionData.empty.copy(
+                      journeyStatus = Some(fillingOutReturn)
+                    )
+                  )
+                )
+              )
+            )
+          }
+
+          val result = performAction()
+          status(result) shouldBe OK
+
+          val doc: Document = parse(contentAsString(result))
+          doc.title shouldBe "Report and pay Capital Gains Tax on UK property - GOV.UK"
```

Comment:
can we assert against `messagefromMessageKey("service.title")` here?

Reply:
done